apple/testing/default_runner/macos_test_runner.template.sh

```diff
@@ -116,9 +116,15 @@ if [[ "${COVERAGE:-}" -eq 1 ]]; then
 fi
 
 XCTESTRUN_ENV=""
+COVERAGE_PRODUCE_JSON=false
 for SINGLE_TEST_ENV in ${TEST_ENV//,/ }; do
   IFS== read key value <<< "$SINGLE_TEST_ENV"
   XCTESTRUN_ENV+="<key>$(escape "$key")</key><string>$(escape "$value")</string>"
+  if [[ "${COVERAGE:-}" -eq 1 ]]; then
+    if [[ "$key" = "COVERAGE_PRODUCE_JSON" && "$value" = "TRUE" ]]; then
+      COVERAGE_PRODUCE_JSON=true
+    fi
+  fi
```

Comment:
I don't think we should have to parse this, I think this would just be set in the environment with `--test_env`?